patches/server/0005-Paper-config-files.patch

```diff
@@ -3543,6 +3545,76 @@ index 0000000000000000000000000000000000000000..90071acf589fb2fa1c3480883165b429
 +    }
 +
 +}
+diff --git a/src/main/java/io/papermc/paper/configuration/transformation/global/versioned/V30_VelocitySecretFile.java b/src/main/java/io/papermc/paper/configuration/transformation/global/versioned/V30_VelocitySecretFile.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..2e0a7ff0cd92e3e3be3ae09bf2d781f5cfd5ff26
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/configuration/transformation/global/versioned/V30_VelocitySecretFile.java
+@@ -0,0 +1,64 @@
++package io.papermc.paper.configuration.transformation.global.versioned;
++
++import com.mojang.logging.LogUtils;
++import org.checkerframework.checker.nullness.qual.Nullable;
++import org.slf4j.Logger;
++import org.spongepowered.configurate.ConfigurateException;
++import org.spongepowered.configurate.ConfigurationNode;
++import org.spongepowered.configurate.NodePath;
++import org.spongepowered.configurate.transformation.ConfigurationTransformation;
++import org.spongepowered.configurate.transformation.TransformAction;
++
++import java.io.IOException;
++import java.nio.file.Files;
++import java.nio.file.Path;
++import java.nio.file.StandardOpenOption;
++import java.util.Optional;
++import java.util.function.Predicate;
++
++import static org.spongepowered.configurate.NodePath.path;
++
++public class V30_VelocitySecretFile implements TransformAction {
++    private static final Logger LOGGER = LogUtils.getLogger();
++    private static final int VERSION = 30;
++    private static final V30_VelocitySecretFile INSTANCE = new V30_VelocitySecretFile();
++    private static final String DEFAULT_SECRET_FILE = "forwarding.secret";
++    private static final NodePath PATH = path("proxies", "velocity");
++    private static final String LEGACY_SECRET_PATH = "secret";
++    private static final String SECRET_FILE_PATH = "secret-file";
++
++    private V30_VelocitySecretFile() {
++    }
++
++    public static void apply(ConfigurationTransformation.VersionedBuilder builder) {
++        builder.addVersion(VERSION, ConfigurationTransformation.builder().addAction(PATH, INSTANCE).build());
++    }
++    @Override
++    public Object @Nullable [] visitPath(final NodePath path, final ConfigurationNode value) throws ConfigurateException {
++        if (value.hasChild(LEGACY_SECRET_PATH)) {
++            final String secretFile = getSecretFile(value);
```

Comment:
I think we can assume that since this versioned migration is running, the secret file path isn't set in the config right?